src/ucs/async/async.c

```diff
@@ -332,12 +332,19 @@ ucs_status_t ucs_async_dispatch_handlers(int *handler_ids, size_t count,
 ucs_status_t ucs_async_dispatch_timerq(ucs_timer_queue_t *timerq,
                                        ucs_time_t current_time)
 {
-    size_t max_timers, num_timers = 0;
+    size_t num_timers = 0;
+    ucs_status_t status;
+    size_t max_timers, size;
     int *expired_timers;
     ucs_timer_t *timer;
 
-    max_timers     = ucs_max(1, ucs_timerq_size(timerq));
-    expired_timers = ucs_alloca(max_timers * sizeof(*expired_timers));
+    max_timers = ucs_min(1024, ucs_max(1, ucs_timerq_size(timerq)));
```

Comment:
why 1024 is a minimum?

Reply:
This is to cap to allow batching, if we end-up having a really huge number of timers to expire. Previously we were using alloca() with 1200B limit, or 1200/4=300 timers, so we are quadrupling it, but true we could also removing it altogether, you want to remove it?

Reply:
yes, i'd prefer to keep previous behavior